### PR TITLE
gnrc_sock: set remote network interface implicitly

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
When there is only one interface we are simplifying a lot for the users
if the interface is set implicitly.

### Issues/PRs references
"Fixes" the problems encountered in #8457.

### CODE REVIEW

File: sys/net/gnrc/sock/include/gnrc_sock_internal.h [hunk 1]
```diff
@@ -96,6 +97,25 @@ static inline bool gnrc_ep_addr_any(const sock_ip_ep_t *ep)
...
 
+/**
+ * @brief   Initializes a sock end-point from a given and sets the network
+ *          interface implicitly if there is only one potential interface.
+ * @internal
+ */
+static inline void gnrc_ep_set(sock_ip_ep_t *out, const sock_ip_ep_t *in,
+                               size_t in_size)
+{
+    memcpy(out, in, in_size);
```
Comment (bergzand) [inline sys/net/gnrc/sock/include/gnrc_sock_internal.h]: Is there a reason as not to fix the size to `sizeof(sock_ip_ep_t)` here (and remove the extra argument with it)?

File: sys/net/gnrc/sock/udp/gnrc_sock_udp.c [hunk 2]
```diff
@@ -278,6 +279,14 @@ ssize_t sock_udp_send(sock_udp_t *sock, const void *data, size_t len,
     }
     else {
         rem = (sock_ip_ep_t *)remote;
+#if GNRC_NETIF_NUMOF == 1
+        /* set interface implicitly */
+        gnrc_netif_t *netif = gnrc_netif_iter(NULL);
+
+        if (netif != NULL) {
+            rem->netif = netif->pid;
```
Comment (bergzand) [inline sys/net/gnrc/sock/udp/gnrc_sock_udp.c]: If I read the code correctly it is possible that this changes the `const sock_udp_ep_t *remote` argument here (first assigning `remote` to `rem` and then changing rem here). 

Besides changing the argument to non const, I don't really have a solution, but I think it is a bit ugly to modify something that is passed as `const`

File: sys/net/gnrc/sock/include/gnrc_sock_internal.h (hunk 1 above)
Comment (miri64) [reply sys/net/gnrc/sock/include/gnrc_sock_internal.h]: Yes, in https://github.com/RIOT-OS/RIOT/pull/8470/files#diff-71ddba2942574ad9a8d73c8de31b2230R126 this is called with `sizeof(sock_udp_ep_t)` ;-).

File: sys/net/gnrc/sock/udp/gnrc_sock_udp.c (hunk 2 above)
Comment (miri64) [reply sys/net/gnrc/sock/udp/gnrc_sock_udp.c]: Mhhh change it to non-const is of the table, but you are right! This is bad. I try to come up with a better solution.

File: sys/net/gnrc/sock/include/gnrc_sock_internal.h (hunk 1 above)
Comment (bergzand) [reply sys/net/gnrc/sock/include/gnrc_sock_internal.h]: Never mind then, It all looked like the same `sock_something_ep_t` from here :)

File: sys/net/gnrc/sock/udp/gnrc_sock_udp.c (hunk 2 above)
Comment (miri64) [reply sys/net/gnrc/sock/udp/gnrc_sock_udp.c]: Done

**Comment by bergzand** [comment]
Tested, works. Please squash.

**Comment by rfuentess** [comment]
Tested. Works like a charm.

**Comment by miri64** [comment]
Squashed. Sorry for the delay.

**Comment by bergzand** [comment]
Just to be sure, Travis/cppcheck complaints look not completely valid to me

**Comment by miri64** [comment]
Yes, but I think it doesn't hurt to copy the cpp-check suppression from ll246ff ;-)

**Comment by miri64** [comment]
Done

**Comment by bergzand** [comment]
I think the suppression needs to be above line 255, travis is still complaining

**Comment by miri64** [comment]
> I think the suppression needs to be above line 255, travis is still complaining

Arghs that comes from editing on master first and then changing to the branch ^^". Fixed and amended immediately.

**Comment by bergzand** [comment]
And now the cppcheck from murdock started to complain. Looks like the same cppcheck exception should be added to the comment at l143 of `sys/net/gnrc/sock/ip/gnrc_sock_ip.c`

**Comment by miri64** [comment]
Fixed and amended immediately.

**Pull request merged**